code/modules/mob/living/carbon/xenomorph/powers.dm

```diff
@@ -401,7 +401,7 @@ GLOBAL_LIST_INIT(acid_spray_hit, typecacheof(list(/obj/structure/barricade, /obj
 	to_chat(src, "<span class='notice'>You start salvaging plasma from [target].</span>")
 
 	while(target.plasma_stored && plasma_stored >= xeno_caste.plasma_max)
-		if(!do_after(src, salvage_delay, TRUE, 5, BUSY_ICON_HOSTILE) || !check_state())
+		if(!do_after(src, salvage_delay, TRUE, null, USER_ICON_GENERIC) || !check_state() || QDELETED(target))
```

Comment:
Same here

Reply:
Same as above.. even though it's a corpse.